Kudu.Core/SiteExtensions/SiteExtensionStatus.cs

```diff
@@ -117,6 +119,11 @@ public bool IsTerminalStatus()
                 || string.Equals(Constants.SiteExtensionProvisioningStateCanceled, ProvisioningState, StringComparison.OrdinalIgnoreCase);
         }
 
+        public bool IsRequireRestart()
+        {
+            return IsSiteExtensionRequireRestart(this, _rootPath);
```

Comment:
"IsSiteExtensionRequireRestart" is for shared the logic with "IsAnyInstallationRequireRestart"

when perform GET, we use "IsAnyInstallationRequireRestart" to check if there is any newly installed extension required to restart, if we find out, we will return header to GEO to trigger restart. and "IsAnyInstallationRequireRestart" is static, that is why "IsSiteExtensionRequireRestart" is static